lib/windowsexec/syscall_windows.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package windowsexec
+
+// this package is called windowsexec rather than just `windows` because
+// calling the package `windows` causes `mkwinsyscall` to generate code
+// without the appropriate package name.
+
+import (
+	"strings"
+	"syscall"
+	"time"
+	"unsafe"
+
+	"github.com/gravitational/trace"
+	"golang.org/x/sys/windows"
+)
+
+//go:generate go run golang.org/x/sys/windows/mkwinsyscall -output zsyscall_windows.go syscall_windows.go
+//sys	shellExecuteExW(info *shellExecuteInfoW) (wasSuccess bool) = shell32.ShellExecuteExW
```

Comment:
I think `shellExecuteExW(info *shellExecuteInfoW) (err error) = shell32.ShellExecuteExW` would work better. Then if the return is 0 (`FALSE`), the last error will be returned without you having to get it yourself.

From mkwinsyscall docs:
```
Each function that returns err needs to supply a condition, that
return value of winapi will be tested against to detect failure.
This would set err to windows "last-error", otherwise it will be nil.
The value can be provided at end of //sys declaration, like
//sys LoadLibrary(libname string) (handle uint32, err error) [failretval==-1] = LoadLibraryA
and is [failretval==0] by default.
```

Reply:
7a59f9be5d